boards/google/quincy/google_quincy_defconfig

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2023 Google Inc
+# SPDX-License-Identifier: Apache-2.0
+
+# Serial Drivers
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+
+# Console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# GPIO Controller
+CONFIG_GPIO=y
+
+# Clock Controller
+CONFIG_CLOCK_CONTROL=y
+
+# Pin Controller
+CONFIG_PINCTRL=y
+
```

Comment:
```sugggestion
```

Reply:
Think this was a white space comment and has been resolved, please re-open if I misunderstood

Reply:
Not a whitespace comment, the comment was to remove the whole highlighted section

Reply:
I need this configuration defined, but can move it to the app instead if more appropriate. Is the concern that pin control isn't strictly board related? I'd like to understand the reasoning here. Thanks

Reply:
Removed for now

Reply:
Then the drivers for this board are wrong, **drivers** should select this Kconfig, not an application or board

Reply:
> I need this configuration defined, but can move it to the app instead if more appropriate.

As @nordicjm mentioned `PINCTRL` is selected when you use driver that requires pinctrl, e.g. [UART driver](https://github.com/zephyrproject-rtos/zephyr/blob/da91dbdd528194c68e7bfea0029541f4335ed9b6/drivers/serial/Kconfig.npcx#L14). You don't need to enable it neither in board nor in application. If it doesn't work, then there is a bug somewhere.

Reply:
Thanks for the explanation and input, I've copied Patryk on an internal bug against the app and added the SOC/driver provider for input. If this results in driver changes as expected, we'll be sure to upstream. Resolving as this is now moved outside the scope of this PR since CONFIG_PINCTRL has been removed from it.